indico/modules/events/abstracts/operations.py

```diff
@@ -29,13 +29,18 @@
 from indico.modules.events.abstracts.notifications import send_abstract_notifications
 from indico.modules.events.contributions.operations import delete_contribution
 from indico.modules.events.logs.models.entries import EventLogRealm, EventLogKind
+from indico.modules.events.tracks.models.tracks import Track
 from indico.modules.events.util import set_custom_fields
 from indico.util.date_time import now_utc
 from indico.util.fs import secure_filename
 
 
 def create_abstract(event, abstract_data, custom_fields_data=None):
     abstract = Abstract(event_new=event, submitter=session.user)
+    if isinstance(abstract_data['submitted_for_tracks'], Track):
+        abstract_data['submitted_for_tracks'] = {abstract_data['submitted_for_tracks']}
```

Comment:
Instead of modifying the dict passed to the function I'd rather assign `tracks = ...` here and then just set `abstract.submitted_for_tracks = tracks` after the `populate_from_dict()`